studies/helpers.py

```diff
@@ -85,11 +85,14 @@ def exit(self, response, frame_data: dict, *args, **kwargs):
             response: Response Model.
             frame_data: dictionary of frame data.
         """
-        withdrawal = frame_data.get("withdrawal", False)
+        withdrawal = frame_data.get("withdrawal", None)
         if withdrawal:
-            response.videos.filter(
-                is_consent_footage=False
-            ).delete()  # Should trigger a pre-delete hook.
+            for video in response.videos.filter(is_consent_footage=False):
+                video.delete(delete_in_s3=True)
+        else:
```

Comment:
this warning will also be triggered whenever withdrawal=False - we want to check explicitly for None (or just for that key not existing), and otherwise trigger the withdrawal actions iff appropriate

Reply:
got it